### PR TITLE
Kozea/WeasyPrint#106: handle rtl ol counter

### PR DESCRIPTION
**Pull request by malnajdi**
This pull request is related to the issue, #106 

This is an early attempt to fixing the issue of ordered list counter not respecting the direction when it's rtl.

### CODE REVIEW

**Comment by malnajdi** [comment]
I am not sure how can i add test to this pull request

File: weasyprint/formatting_structure/build.py [hunk 1]
```diff
@@ -311,9 +319,11 @@ def marker_to_box(element, state, parent_style, style_for, get_image_from_uri,
         # See https://drafts.csswg.org/css-pseudo-4/#marker-pseudo
         marker_box.style['position'] = 'absolute'
         if parent_style['direction'] == 'ltr':
+            marker_box.style['left'] = 'auto'
```
Comment (liZe) [inline weasyprint/formatting_structure/build.py:322]: Why is this line needed?

File: weasyprint/formatting_structure/build.py [hunk 2]
```diff
@@ -311,9 +319,11 @@ def marker_to_box(element, state, parent_style, style_for, get_image_from_uri,
...
             translate_x = properties.Dimension(-100, '%')
         else:
-            translate_x = properties.Dimension(100, '%')
+            marker_box.style['right'] = properties.Dimension(0, 'px')
+            translate_x = properties.Dimension(20, '%')
```
Comment (liZe) [inline weasyprint/formatting_structure/build.py:326]: This 20% value is a bit strange… Why doesn’t the 100% translation work?

File: weasyprint/css/counters.py [hunk 3]
```diff
@@ -280,12 +280,16 @@ def render_value(self, counter_value, counter_name=None, counter=None,
         # Step 6
         return initial
 
-    def render_marker(self, counter_name, counter_value):
+    def render_marker(self, counter_name, counter_value, parent_direction):
```
Comment (liZe) [inline weasyprint/css/counters.py:283]: 👍🏼

File: weasyprint/css/counters.py [hunk 4]
```diff
@@ -295,6 +299,11 @@ def render_marker(self, counter_name, counter_value):
 
         value = self.render_value(counter_value, counter_name=counter_name)
         assert value is not None
+
```
Comment (liZe) [inline weasyprint/css/counters.py:302]: You can put the `suffix = …` from line 298 + the return in an `else` block, that would improve readability in my opinion.

File: weasyprint/formatting_structure/build.py [hunk 5]
```diff
@@ -145,6 +145,15 @@ def element_to_box(element, style_for, get_image_from_uri, base_url,
 
     marker_boxes = []
     if style['display'] == 'list-item':
+        # This should handle rtl direction
```
Comment (liZe) [inline weasyprint/formatting_structure/build.py:148]: Why is this code needed?

File: weasyprint/css/counters.py [hunk 6]
```diff
@@ -295,6 +299,11 @@ def render_marker(self, counter_name, counter_value):
...
+        if parent_direction == 'rtl':
+            suffix = symbol(counter['suffix'] or ('string', ' .'))
```
Comment (liZe) [inline weasyprint/css/counters.py:304]: Having ' .' instead of '. ' is an easy fix, but I’m not sure about how the specification handles this. We should read it, and maybe implement [`marker-side`](https://drafts.csswg.org/css-lists-3/#marker-side) to have a clean solution.

**Review by liZe — CHANGES_REQUESTED** [review]
The overall idea is good, and it could be merged (with the translation hack removed :wink:). But I’d really like to know the specification handles the order of prefix / value / suffix and the space before / after the full stop…

File: weasyprint/css/counters.py (hunk 6 above)
Comment (malnajdi) [reply weasyprint/css/counters.py:304]: I did not know marker-side specification was there. I am back with the specification but, thank you for pointing that out. I will try to implement it.

File: weasyprint/formatting_structure/build.py (hunk 5 above)
Comment (malnajdi) [reply weasyprint/formatting_structure/build.py:148]: It's not needed, i will remove it.

File: weasyprint/formatting_structure/build.py (hunk 2 above)
Comment (malnajdi) [reply weasyprint/formatting_structure/build.py:326]: I am not sure why 100% did not work. I know 20 is really strange and i will investigate more.

File: weasyprint/formatting_structure/build.py (hunk 1 above)
Comment (malnajdi) [reply weasyprint/formatting_structure/build.py:322]: After applying this patch it seems the only way for `direction: ltr` to work is to set left to auto.

File: weasyprint/css/counters.py (hunk 4 above)
Comment (malnajdi) [reply weasyprint/css/counters.py:302]: Will do.

**Comment by liZe** [comment]
I’ll close this PR for now, feel free to reopen if you find the time to work on this!
